src/frontends/tensorflow_common/src/op/select.cpp

```diff
@@ -54,10 +78,61 @@ OutputVector translate_select_op(const NodeContext& node) {
     // 1. Either the same shape (in which case the select is elementwise), or
     // 2. condition must be Rank 1 and match over the first dimension, or
     // 3. condition is scalar
-    default_op_checks(node, 3, {"Select"});
+    default_op_checks(node, 3, {"Select"}, true);
     auto condition = node.get_input(0);
     auto x = node.get_input(1);
     auto y = node.get_input(2);
+    auto complex_type_mark_x = as_type_ptr<ComplexTypeMark>(x.get_node_shared_ptr());
+    auto complex_type_mark_y = as_type_ptr<ComplexTypeMark>(y.get_node_shared_ptr());
+    auto x_rank = compute_subgraph_scalar_rank(x, element::i32);
+
+    if (complex_type_mark_x || complex_type_mark_y) {
+        x = complex_type_mark_x->input_value(0);
+        y = complex_type_mark_y->input_value(0);
+
+        auto gather_index_real = make_shared<v0::Constant>(element::i32, Shape{}, 0);
+        auto gather_index_imag = make_shared<v0::Constant>(element::i32, Shape{}, 1);
+        auto minus_one = make_shared<v0::Constant>(element::i32, Shape{1}, -1);
+
+        auto x_real = make_shared<v8::Gather>(x, gather_index_real, minus_one)->output(0);
+        auto x_imag = make_shared<v8::Gather>(x, gather_index_imag, minus_one)->output(0);
+
+        auto y_real = make_shared<v8::Gather>(y, gather_index_real, minus_one)->output(0);
+        auto y_imag = make_shared<v8::Gather>(y, gather_index_imag, minus_one)->output(0);
+        x_rank = compute_subgraph_scalar_rank(x_real, element::i32);
```

Comment:
don't think we need to extract real and imag parts. We can work with shape of `x` to compute `new_cond_shape`